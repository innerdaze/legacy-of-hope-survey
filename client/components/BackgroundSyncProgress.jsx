import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Box from 'grommet/components/Box'
import Meter from 'grommet/components/Meter'
import Value from 'grommet/components/Value'

class BackgroundSyncProgress extends Component {
  render() {
    return this.props.isSyncing && (
      <Box>
        <Meter
          type='bar'
          max={1}
          fill={true}
          value={this.props.progress}
          label={<Value
            units='%'
            responsive={true}
            value={Math.floor(this.props.progress * 100, 2)}
            />}
          />
      </Box>
    ) || null
  }
}

BackgroundSyncProgress.propTypes = {
  isSyncing: PropTypes.bool.isRequired,
  progress: PropTypes.number.isRequired
}

BackgroundSyncProgress.defaultProps = {
  isSyncing: false,
  progress: 0
}

export default BackgroundSyncProgress
