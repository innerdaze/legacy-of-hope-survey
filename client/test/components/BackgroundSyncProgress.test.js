import React from 'react'
import BackgroundSyncProgress from '../../components/BackgroundSyncProgress'

let wrapper, progress

describe('Testing on BackgroundSyncProgress...', () => {
  beforeEach(() => {
    progress = 0.1
    wrapper = shallow(<BackgroundSyncProgress progress={progress} />)
  })

  describe('BackgroundSyncProgress layout', () => {
    test('Expect BackgroundSyncProgress to have Heading and Meter comp', () => {
      expect(wrapper.find('Box')).toHaveLength(1)
      expect(wrapper.find('Box').find('Heading')).toHaveLength(1)
      expect(wrapper.find('Box').find('Meter')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
