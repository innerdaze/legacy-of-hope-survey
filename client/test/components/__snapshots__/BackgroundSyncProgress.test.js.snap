// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on BackgroundSyncProgress... BackgroundSyncProgress layout Expect BackgroundSyncProgress to have Heading and Meter comp 1`] = `
<Box
  align="stretch"
  announce={false}
  direction="column"
  focusable={true}
  pad="none"
  responsive={true}
  tag="div"
>
  <Heading
    tag="h1"
  >
    Synchronising Data...
  </Heading>
  <Meter
    label={
      <Value
        align="center"
        announce={false}
        units="%"
        value={10}
      />
    }
    max={1}
    size="small"
    type="bar"
    value={0.1}
  />
</Box>
`;
